src/jvmTest/kotlin/bdd/SpaceEngineersCucumberTest.kt

```diff
@@ -100,6 +130,55 @@ class SpaceEngineersCucumberTest {
         )
     }
 
+    private fun observeBlocks(): List<SeBlock> {
+        return environment.observe(ObservationArgs(ObservationMode.BLOCKS)).allBlocks
+    }
+
+    private fun blockToGrind(): SeBlock {
+        return observeBlocks().first { it.id == context.lastNewBlockId }
+    }
+
+    @When("Character grinds to {double} integrity.")
+    fun character_grinds_until_to_integrity(integrity: Double) {
```

Comment:
one of the prepositions is superfluous, "until integrity" should be fine

(It's a detail... are you interested in English language comments?)

Reply:
Sure, happy to get feedback about English too :-) Any constructive feedback is appreciated.